src/pcap/api.go

```diff
@@ -183,9 +199,11 @@ func (api *API) Capture(stream API_CaptureServer) (err error) {
 		return errorf(codes.InvalidArgument, "expected start message, got %v: %w", req.Operation, errUnexpectedMessage)
 	}
 
-	targets, resolveErr := api.resolveAgentEndpoints(opts.Start.Capture, log)
-	if errors.Is(resolveErr, errValidationFailed) {
-		return errorf(codes.InvalidArgument, "capture targets not found: %w", err)
+	targets, resolveErr := api.resolveAgentEndpoints(opts.Start.Request, log)
+	if errors.Is(resolveErr, ErrValidationFailed) {
+		return errorf(codes.InvalidArgument, "capture targets not found: %w", resolveErr)
```

Comment:
I'm struggling to understand what I wrote there :joy:

But looking at it again the confusing bit seems to be that `InvalidArgument` and `NotFound` are two separate things even though they are related. I'm saying the message to the client should be "I can't find what you are looking for" instead of "Your request is malformed" which is expressed in the text but not in the status code.

Reply:
This remark might be a bit late but I will still raise it: Maybe the base error codes should already have the status code assigned to them? After all the bottom most error code usually gives the underlying reason and has the best idea of what exactly went wrong. The mapping seems pretty obvious:
```
ErrValidationFailed  = fmt.Errorf("validation failed")            // InvalidArgument
errTooManyCaptures   = fmt.Errorf("too many concurrent captures") // ResourceExhausted
errDraining          = fmt.Errorf("draining")                     // Unavailable
errUnexpectedMessage = fmt.Errorf("unexpected message")           // InvalidArgument
ErrNoEndpoints       = fmt.Errorf("no matching endpoints found")  // NotFound
ErrResolverUnhealthy = fmt.Errorf("resolver unhealthy")           // Internal
ErrNotAuthorized     = fmt.Errorf("not authorized")               // PermissionDenied (or maybe Unauthenticated, not sure)
```

Reply:
We will tackle this in a different PR in the future.